plugins/org/src/components/Cards/Group/GroupProfile/GroupProfileCard.tsx

```diff
@@ -93,16 +96,42 @@ export const GroupProfileCard = ({
     ?.filter(r => r.type === RELATION_PARENT_OF)
     ?.map(group => group.target.name);
 
+  const displayName = profile?.displayName ?? name;
```

Comment:
We should always fallback to the entity name as the displayName is an optional field. I changed that in multiple locations